src/utils/questionnaireBuilderUtils.ts

```diff
@@ -1,245 +1,331 @@
-    return "Please mark at least one answer as correct by checking the box";
+    return { 
+      isValid: false, 
+      errorType: "noCorrectAnswers" 
+    };
   }
-
+  
+  // Check if pass score is set
   if (questionnaire.passScore === undefined || questionnaire.passScore === null) {
-    return "Please set a required pass score using the slider";
+    return { 
+      isValid: false, 
+      errorType: "noPassScore" 
+    };
   }
-
-  return "Save questionnaire";
+  
+  // All validations passed
+  return { 
+    isValid: true, 
+    errorType: null 
+  };
 };
 
 /**
- * Function to create empty questionnaire object
+ * Checks if a questionnaire form is valid for submission
  * 
- * @returns Empty questionnaire object
+ * @param questionnaire - The questionnaire to validate
+ * @returns Boolean indicating whether the form is valid
  */
-export const createEmptyQuestionnaire = (): Questionnaire => {
-  return {
-    title: "",
-    description: "",
-    questions: [],
-    passScore: 0,
-    tags: [],
-  };
+export const isFormValid = (questionnaire: Questionnaire): boolean => {
+  const { isValid } = validateQuestionnaire(questionnaire);
+  return isValid;
 };
 
 /**
- * Function to check if form is valid for submission
+ * Gets appropriate tooltip message for a questionnaire's validation state
+ * Using only existing localization strings
  * 
- * @param questionnaire Current questionnaire state
- * @returns Boolean indicating if form is valid
+ * @param questionnaire - The questionnaire to get tooltip message for
+ * @param strings - The localization strings object
+ * @returns Localized tooltip message based on validation result
  */
-export const isFormValid = (questionnaire: Questionnaire): boolean => {
-  // Check required fields
-  if (!questionnaire.title || !questionnaire.description) {
-    return false;
+export const getValidationTooltipMessage = (
+  questionnaire: Questionnaire, 
+  strings: any
+): string => {
+  const { errorType } = validateQuestionnaire(questionnaire);
+  
+  if (!errorType) {
+    return ""; // No error
   }
-  if (!questionnaire.questions || questionnaire.questions.length === 0) {
-    return false;
+  
+  // Map error types to existing localized strings
+  switch(errorType) {
+    case "bothEmpty":
+      return strings.newQuestionnaireBuilder.tooltipBothEmpty;
+    case "emptyTitle":
+      return strings.newQuestionnaireBuilder.tooltipEmptyTitle;
+    case "emptyDescription":
+      return strings.newQuestionnaireBuilder.tooltipEmptyDescription;
+    case "noQuestions":
+      return strings.error.questionnaireSaveFailed + ", " + "At least one question is required";
+    case "noCorrectAnswers":
+      return strings.error.questionnaireSaveFailed + ", " + "At least one answer must be marked as correct";
+    case "noPassScore":
+      return strings.error.questionnaireSaveFailed + ", " + "Please set a required score using the slider";
+    default:
+      return strings.error.generic;
```

Comment:
Update cases from ErrorType string to the actual condition being checked as said in previous comment.